lib/semian/net_http.rb

```diff
@@ -67,9 +76,11 @@ def reset_exceptions
     Semian::NetHTTP.reset_exceptions
 
     def raw_semian_options
-      options = semian_configuration
-      options = options.dup unless options.nil?
-      options
+      check_semian_configuration_changed
+      @raw_semian_options ||= begin
+        @raw_semian_options = Semian::NetHTTP.retrieve_semian_configuration(address, port)
+        @raw_semian_options = @raw_semian_options.dup unless @raw_semian_options.nil?
+      end
```

Comment:
But there is no intentional bug? What's the bug? The requirement of having the `proc` return the same value each time isn't a limitation ... if you don't do that and somehow make the `Resource` object call the `proc` each time it wants a `ticket` size, a `error_threshold` or any of the other parameters, it will lead to race conditions.